pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -1457,7 +1458,16 @@ private void makeStreamConsumer(String reason) {
     _segmentLogger.info("Creating new stream consumer for topic partition {} , reason: {}", _clientId, reason);
     _partitionGroupConsumer =
         _streamConsumerFactory.createPartitionGroupConsumer(_clientId, _partitionGroupConsumptionStatus);
-    _partitionGroupConsumer.start(_partitionGroupConsumptionStatus.getStartOffset());
+    _partitionGroupConsumer.start(_currentOffset);
+  }
+  /**
+   * Checkpoints existing consumer before creating a new consumer instance
+   */
+  private void recreateStreamConsumer(String reason) {
+    if (_partitionGroupConsumer != null) {
+      _currentOffset = _partitionGroupConsumer.checkpoint(_currentOffset);
+    }
+    makeStreamConsumer(reason);
```

Comment:
Let's not reuse the `makeStreamConsumer()` for clear separation of the handling
```suggestion
    _segmentLogger.warn("Recreating stream consumer for topic partition: {}, reason: {}", _clientId, reason);
    _currentOffset = _partitionGroupConsumer.checkpoint(_currentOffset);
    closePartitionGroupConsumer();
    _partitionGroupConsumer =
        _streamConsumerFactory.createPartitionGroupConsumer(_clientId, _partitionGroupConsumptionStatus);
    _partitionGroupConsumer.start(_currentOffset);
```

Reply:
Ok. I have made the change.